app/helpers/hosts_helper.rb

```diff
@@ -368,7 +368,7 @@ def host_title_actions(host)
   end
 
   def delete_host_dialog(host)
-    if host.compute?
+    if host.compute? && Setting[:disassociate_vms_before_host_deletion]
       _("Are you sure you want to delete host %s? This will delete the virtual machine and its disks, and is irreversible.") % host.name
     else
       _("Are you sure you want to delete host %s? This action is irreversible.") % host.name
```

Comment:
One more idea, can you actually teach the users here in the dialog?

    _("Are you sure you want to delete host %s? This will delete the VM and its disks, and is irreversible. This behavior can be changed in global settings.") % host.name

and

    _("Are you sure you want to delete host %s? It is irreversible, but VM and its disks will not be deleted. This behavior can be changed in global settings.") % host.name